beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -750,3 +757,371 @@ func (bs *Server) checkSync(ctx context.Context, w http.ResponseWriter) bool {
 	}
 	return true
 }
+
+// ProduceBlockV3 Requests a beacon node to produce a valid block, which can then be signed by a validator. The
+// returned block may be blinded or unblinded, depending on the current state of the network as
+// decided by the execution and beacon nodes.
+// The beacon node must return an unblinded block if it obtains the execution payload from its
+// paired execution node. It must only return a blinded block if it obtains the execution payload
+// header from an MEV relay.
+// Metadata in the response indicates the type of block produced, and the supported types of block
+// will be added to as forks progress.
+func (bs *Server) ProduceBlockV3(w http.ResponseWriter, r *http.Request) {
+	if ok := bs.checkSync(r.Context(), w); !ok {
+		return
+	}
+	segments := strings.Split(r.URL.Path, "/")
+	rawSlot := segments[len(segments)-1]
+	rawRandaoReveal := r.URL.Query().Get("randao_reveal")
+	rawGraffiti := r.URL.Query().Get("graffiti")
+	rawSkipRandaoVerification := r.URL.Query().Get("skip_randao_verification")
+
+	if rawSlot == "" {
+		errJson := &http2.DefaultErrorJson{
+			Message: "slot is required",
+			Code:    http.StatusBadRequest,
+		}
+		http2.WriteError(w, errJson)
+		return
+	}
+	slot, err := strconv.ParseUint(rawSlot, 10, 64)
+	if err != nil {
+		errJson := &http2.DefaultErrorJson{
+			Message: "slot is invalid: " + err.Error(),
+			Code:    http.StatusBadRequest,
+		}
+		http2.WriteError(w, errJson)
+		return
+	}
+	var randaoReveal []byte
+	if rawSkipRandaoVerification == "true" {
+		randaoReveal = primitives.PointAtInfinity
+	} else {
+		rr, err := hexutil.Decode(rawRandaoReveal)
```

Comment:
it only returns a bool, i need the value, doesn't seem to decode

Reply:
Ok that's true... In that case maybe let's return bytes from `shared.ValidateHex`. What do you think? And inside the function use `hexutil.Decode` instead of `bytesutil.IsHex`

Reply:
yeah let me add that in a separate PR when fixing the v2 conversion and using the v1alpha1 ssz